### DISCUSSION TITLE
[V5] sum with two column.

### DISCUSSION THREAD

**Discussion by pk4all** [adonisjs/core#3055] [Help]
Hello Team,

I want to use 
```javascript 
q.sum('time_in_mintues * occurance').as('totalOprationTime')

```
like raw query sum(col1 * col2)
How to achieve this?
Thnaks.

**Comment by pk4all** [comment]
@thetutlage please suggest me a solution.

**Reply by pk4all** [reply]
Hi @RomainLanz any comment on this?

**Comment by McSneaky** [comment]
```ts
query.sum('(time_in_mintues * occurance) AS totalOprationTime')
```

Or
```ts
query.raw('SUM(time_in_mintues * occurance) AS totalOprationTime')
```

Might differ based on what DB you are using

**Reply by pk4all** [reply]
Thanks.

**Reply by pk4all** [reply]
> ```ts
> query.sum('(time_in_mintues * occurance) AS totalOprationTime')
> ```
> 
> Or
> 
> ```ts
> query.raw('SUM(time_in_mintues * occurance) AS totalOprationTime')
> ```
> 
> Might differ based on what DB you are using

Not working with "withAggregate"

My query is 
```
q.withAggregate('operations',otq=>{
                            otq
                            .pojo<{totalOprationTime:number}>()
                            .sum('time_in_mintues * occurance').as('totalOprationTime')
                          })
```
@McSneaky  Please help me.

**Reply by pk4all** [reply]
> > ```ts
> > query.sum('(time_in_mintues * occurance) AS totalOprationTime')
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Or
> > ```ts
> > query.raw('SUM(time_in_mintues * occurance) AS totalOprationTime')
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Might differ based on what DB you are using
> 
> Not working with "withAggregate"
> 
> My query is
> 
> ```
> q.withAggregate('operations',otq=>{
>                             otq
>                             .pojo<{totalOprationTime:number}>()
>                             .sum('time_in_mintues * occurance').as('totalOprationTime')
>                           })
> ```
> 




@McSneaky @RomainLanz any solution for my problom.

**Comment by lucasKuratani** [comment]
You can use `Database.raw` inside `query.select`. It also works after subqueries.

Example:

`Database.from('YOURTABLE').select(Database.raw('(time_in_mintues * occurance) AS totalOprationTime'))`

**Reply by pk4all** [reply]
> You can use `Database.raw` inside `query.select`. It also works after subqueries.
> 
> Example:
> 
> `Database.from('YOURTABLE').select(Database.raw('(time_in_mintues * occurance) AS totalOprationTime'))`

How can I use this with ```withAggregate``` ?
```
q.withAggregate('operations',otq=>{
                            otq
                            .pojo<{totalOprationTime:number}>()
                            .sum('time_in_mintues * occurance').as('totalOprationTime')
                          })
```
